### PR TITLE
Implemented basic cache for /maps API

### PR DESCRIPTION
**Pull request by Bux42**
Now using a cache for the maps route, where we had this big mongo query firing every time the /maps route was called

The request was taking ~4 seconds
And now it is ~100 ms

### CODE REVIEW

**Review by davidbmaier — APPROVED** [review]
The logic for dynamically updating the map cache could probably be moved out of the db handler (to make it a little cleaner), but it looks fine to me already.

File: server/src/cache.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                lastUpdate: metadata.date,
+            });
+        }
+        dbCache.set('maps', cachedMaps);
+    });
+};
+
+export const deleteReplay = (replay?: any) => {
+    if (replay) {
+        getMapsCache().then((cachedMaps: any) => {
```
Comment (MKuijpers) [inline server/src/cache.ts:45]: Can you rewrite this using await for consistency?

File: server/src/cache.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    logInfo('mapsCache: Initializing cache');
+
+    const maps = await db.getMapsStats();
+    dbCache.set('maps', maps);
+    return maps;
+};
+
+export const addReplay = (metadata: any) => {
+    getMapsCache().then((cachedMaps: any) => {
```
Comment (MKuijpers) [inline server/src/cache.ts:23]: Can you rewrite this using await for consistency?

File: server/src/cache.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                    logInfo('mapsCache: deleteReplayById: map play count is 0, deleting map from cache');
+                    cachedMaps.splice(cachedMaps.indexOf(mapCacheMatch), 1);
+                } else {
+                    logInfo('mapsCache: deleteReplayById: decrementing map play count');
+                }
+                dbCache.set('maps', cachedMaps);
+            }
+        });
+    } else {
+        logInfo('mapsCache: deleteReplayById: replay not found');
```
Comment (MKuijpers) [inline server/src/cache.ts:59]: This log should say that no replay was passed to the method if it's logged if `replay` is not defined. Is there even a reason why this replay parameter is optional? It seems kind of useless to call a deleteReplay method without a replay. Making it required would also remove the `if(replay)` and this log statement

**Review by MKuijpers — COMMENTED** [review]
Left some small comments. Should be easy to fix

**Pull request merged**